examples/seismic/model.py

```diff
@@ -377,14 +405,22 @@ def __init__(self, origin, spacing, shape, space_order, vp, nbpml=20,
         else:
             self.theta = 0
 
-        if phi is not None:
+        if phi is not None and self.grid.dim>2:
             if isinstance(phi, np.ndarray):
                 self.phi = Function(name="phi", grid=self.grid, space_order=space_order)
                 initialize_function(self.phi, phi, self.nbpml)
             else:
                 self.phi = phi
         else:
-            self.phi = 0
+            self.phi = None
+
+    def physical_params(self, **kwargs):
+        """
+        Return all set physical parameters
+        """
+        known = [getattr(self, i) for i in self._physical_parameters]
+        return {i.name: kwargs.get(i.name, i) or i for i in known
+                if isinstance(i, Function)}
```

Comment:
It took me a moment to understand why you're checking if it's a Function. I guess it's much more obvious if you check if it's not None

Reply:
No, want to check it is a function as some of these parameters can be constant values and should not be passed as argument to `apply`.